File: Part 1 - Data Preprocessing/Section 2 -------------------- Part 1 - Data Preprocessing --------------------/prep.R

dataset = read.csv("Data.csv")

# take care of missing data
dataset$Age = ifelse(
  is.na(dataset$Age),
  ave(dataset$Age, FUN = function(x) mean(x, na.rm = TRUE)),
  dataset$Age
)

dataset$Salary = ifelse(
  is.na(dataset$Salary),
  ave(dataset$Salary, FUN = function(x) mean(x, na.rm = TRUE)),
  dataset$Salary
)

# enconde categorical data
dataset$Country = factor(
  dataset$Country, levels = c('France', 'Spain', 'Germany'), labels = c(1, 2, 3)
) # this labels are non-numeric but labels, even if they contain numbers

dataset$Purchased = factor(
  dataset$Purchased, levels = c('No', 'Yes'), labels = c(0, 1)
)


# split the dataset into training and testing sets
# install.packages('catools')
set.seed(123)
split = sample.split(dataset$Purchased, SplitRatio = 0.8) # SplitRatio is the percentaje of the dataset that goes to the training set
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)


# scale the training sets
training_set[, 2:3] = scale(training_set[, 2:3])
test_set[, 2:3] = scale(test_set[, 2:3]) # we are using 2:3 to exclude column 1, which is non-numeric

File: Part 1 - Data Preprocessing/Section 2 -------------------- Part 1 - Data Preprocessing --------------------/prep.py
# -*- coding: utf-8 -*-
"""
Spyder Editor

This is a temporary script file.
"""

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd


#importing the dataset
\
dataset = pd.read_csv("Data.csv")

x = dataset.iloc[:,:-1].values
y = dataset.iloc[:,3].values


# taking care of missing data
from sklearn.preprocessing import Imputer
inputer = Imputer() # inspect the element to see the default parameters/values used
inputer.fit(x[:, 1:3])
x[:, 1:3] = inputer.transform(x[:, 1:3]) # replace the missing data by the mean of the column


# encode categorical data -----------------------------------------------------
from sklearn.preprocessing import LabelEncoder, OneHotEncoder

x[:, 0] = LabelEncoder().fit_transform(x[:, 0]) # encode the first colum (Countries)
# problem here is that we are giving each country a secuential number, which would make them seem to have some order/priority
#so instead we are going to use dummy encoding:
''' [0] : array of columns considered'''
x = OneHotEncoder(categorical_features = [0]).fit_transform(x).toarray()

# be sure to use a new labelenconder for each transform, since they are fitted
y = LabelEncoder().fit_transform(y) # encode the first colum (Countries)
# -----------------------------------------------------------------------------

# splt the dataset into training and testing
from sklearn.model_selection import train_test_split
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.2, random_state=0)


from sklearn.preprocessing import StandardScaler
x_train = StandardScaler().fit_transform(x_train)
x_test = StandardScaler().fit_transform(x_test)



























